buy_together_app/tests/test_login.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+from unittest.mock import Mock
+from buy_together_app import views
+from django.shortcuts import render, redirect
+from django.test import RequestFactory
+
+
+@pytest.fixture
+def authenticate_mock(monkeypatch):
+    mock = Mock()
+    monkeypatch.setattr(views, views.authenticate.__name__, mock)
+    return mock
+
+
+@pytest.fixture
+def login_mock(monkeypatch):
+    mock = Mock()
+    monkeypatch.setattr(views, views.login.__name__, mock)
+    return mock
+
+
+@pytest.fixture
+def redirect_mock(monkeypatch):
+    mock = Mock()
+    monkeypatch.setattr(views, views.redirect.__name__, mock)
+    mock.return_value = redirect('Main Page')
+    return mock
+
+
+@pytest.fixture
+def messages_mock(monkeypatch):
+    mock = Mock()
+    monkeypatch.setattr(views, "messages", mock)
+    return mock
+
+
+@pytest.fixture
+def render_mock(monkeypatch, get_request):
+    mock = Mock()
+    monkeypatch.setattr(views, views.render.__name__, mock)
+    mock.return_value = render(get_request, 'buy_together_app/login.html')
+    return mock
+
+
+@pytest.fixture
+def post_method_mock():
```

Comment:
Should be something like: `request_with_post_mock` (and similar logic for the other with non-post), as the object that you're returning is a mock for the request rather than a mock for the API method.